lib/spack/spack/util/web.py

```diff
@@ -38,7 +39,7 @@
     from html.parser import HTMLParser
 
     # Also, HTMLParseError is deprecated and never raised.
-    class HTMLParseError:
+    class HTMLParseError(BaseException):
```

Comment:
Minor thing: I think it's better to inherit from `Exception`, `BaseException` is usually not meant for user defined classes. See [here](https://docs.python.org/2/library/exceptions.html).